mne/preprocessing/ica.py

```diff
@@ -2079,3 +2079,233 @@ def _band_pass_filter(ica, sources, target, l_freq, h_freq, verbose=None):
+        0.6 and 0.95. In the original proposal, lower values are considered,
+        but this is not yet implemented.
+        If list of floats: search for the best map in the specified range of
+        correlation strengths.
+        If float > 0: select ICs correlating better than this.
+        If float > 1: use find_outliers to identify ICs within subjects (not in
+        original Corrmap)
+        Defaults to "auto".
+    name : str
+        Categorised ICs are stored in a default dictionary "labels". This
+        parameter gives the key under which found ICs will be stored.
+        Defaults to "bads".
+    plot : bool
+        Should constructed template and selected maps be plotted?
+    ch_type : 'mag' | 'grad' | 'planar1' | 'planar2' | 'eeg'
+            The channel type to plot. Defaults to 'eeg'.
+
+    Returns
+    -------
+    new_icas  :  list
+        Returns a list of fitted ICs, enrichened with the indices of the
+        selected maps in the field specified by the name keyword.
+    """
+
+    def vcorrcoef(X, y):  # vectorised correlation, by Jon Herman
+        Xm = np.reshape(np.mean(X, axis=1), (X.shape[0], 1))
+        ym = np.mean(y)
+        r_num = np.sum((X - Xm) * (y - ym), axis=1)
+        r_den = np.sqrt(np.sum((X - Xm) ** 2, axis=1) * np.sum((y - ym) ** 2))
+        r = r_num / r_den
+        return r
+
+    def get_ica_map(ica, components=None):
+        if components is None:
+            components = range(ica.n_components_)
+        maps = np.dot(ica.mixing_matrix_[:, components].T,
+                      ica.pca_components_[:ica.n_components_])
+        return maps
+
+    def find_max_corrs(all_maps, target, threshold):
+        all_corrs = [vcorrcoef(subj, target) for subj in all_maps]
+        abs_corrs = [np.abs(a) for a in all_corrs]
+        corr_polarities = [np.sign(a) for a in all_corrs]
+
+        if threshold <= 1:
+            max_corrs = [list(np.nonzero(s_corr > threshold)[0])
+                         for s_corr in abs_corrs]
+        else:
+            max_corrs = [list(find_outliers(s_corr, threshold=threshold))
+                         for s_corr in abs_corrs]
+
+        am = [l[i] for l, i_s in zip(abs_corrs, max_corrs)
+              for i in i_s]
+        median_corr_with_target = np.median(am)
+
+        polarities = [l[i] for l, i_s in zip(corr_polarities, max_corrs)
+                      for i in i_s]
+
+        maxmaps = [l[i] for l, i_s in zip(all_maps, max_corrs)
+                   for i in i_s]
+
+        try:
+            newtarget = np.zeros(maxmaps[0].size)
+            for maxmap, polarity in zip(maxmaps, polarities):
+                newtarget += maxmap * polarity
+
+            newtarget /= len(maxmaps)
+
+            sim_i_o = np.abs(np.corrcoef(target, newtarget)[1, 0])
+
+            return newtarget, median_corr_with_target, sim_i_o, max_corrs
+        except IndexError:
+            return [], 0, 0, []
```

Comment:
In this case, there are 3 options, and the empty list is never a problem in any of them. Either we are in the 1st iteration and using max correlations, in which case we always have at least one IC - the template itself. Or we use find_outliers, in which if we have a 0, case we ``return`` earlier, with a proper failure message. Or we are in the 2nd iteration and using the "auto" method, and somehow do not manage to find any candidates (which should be rather unlikely) - in which case we iterate over the empty list in the next step, which won't crash, but the user will not be explicitly informed that no map has been selected. This is extremely improbable (as at the very least, the template should be selected), but I can add a line to catch that case, if you think that'd be sufficient.

Are you okay with the ```except:``` in line 2293?